pandas/tests/io/parser/test_common.py

```diff
@@ -64,13 +64,14 @@ def _set_noconvert_columns(self):
         "parse_dates": parse_dates,
         "delimiter": ",",
     }
+    parser.engine = "c"
     parser._engine = MyCParserWrapper(StringIO(data), **parser.options)
 
     result = parser.read()
     tm.assert_frame_equal(result, expected)
 
 
-def test_empty_decimal_marker(all_parsers):
+def test_empty_decimal_marker(all_parsers, pyarrow_xfail):
```

Comment:
instead of this, why don't you just define a new fixture (all_parsers_xpyarrow) or something and then just change the inputs to all of these functions), otherwise this is awkward and unmaintanable.

Reply:
And this `all_parsers_xpyarrow` fixture would be "all parsers but without pyarrow"? 

If so, how is that necessarily more maintainable? You still need to update each test function that doesn't support pyarrow to change `all_parsers` to `all_parsers_xpyarrow`, while with the current approach you need to update this to `all_parsers, pyarrow_xfail`. That doesn't seem a big difference?

Reply:
I might have a way to make this clearer.

at the moment the fixture is xfailing tests explicitly and not using an xfail marker.

I have changed this but there are some tests that seem to lock up, so also need a skip fixture.

Am working through the failing/timing out tests. will do a wip commit for discusssion. (just revert if not happy)

Reply:
hmm, frequent lock-ups on a full test run of tests/io/parser/ so several xfails changed to skips.

hopefully, this won't time out and with the decorators makes it easier to review for functionality that should be working.

Reply:
macOS py37_macos didn't time out this time around, but looks like Windows and linux still have a problem.

Reply:
@simonjayhawkins yeah i had problems with it getting stuck on my computer too. I think the tests are actually failing but pytest just gets stuck or something. I think it should go away once everything is xfailed

Reply:
Also @jreback @jorisvandenbossche I think pyarrow_xfail makes more logical sense as apposed to all_parsers_xpyarrow and is shorter. Besides, you are still going have to xfail every case that fails one by one.